ArticleShapeExtractor/ExtractArticleShapes.jsx

```diff
@@ -2,18 +2,18 @@
 
 try {
     if (app.documents.length === 0) {
-        throw new NoDocumentOpenedError(null, $.fileName, $.line);
+        throw new NoDocumentOpenedError(null, $.fileName, $.line, Severity.LEVELS.NOTICE);
```

Comment:
Is it really necessary to do this by throwing an exception? I am not a big fan of using exceptions for regular code flow.

Reply:
I didn't like my very last enhancement. So I partially have rolled it back in 9708b759dab7a09b72e02207157f3d706713b0f5. I think there is a say for both. What I like about exceptions it is easier for test scripts (to be added later?) to set an expectation by checking the error type.

Reply:
In the end I don't mind too much as I will not be the maintainer of this code. [This](https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why) describes why using exceptions may be a problem. To me the main thing is that it becomes more difficult to reason about the code as you always have be aware that an exception can be raised.

Reply:
Discussed and agreed. Error handling will be picked up in a separate PR, after the UXP port.